### DISCUSSION TITLE
Mac version is using outdated frameworks destined for deprecation: CGDisplayStream and CGWindowList

### DISCUSSION THREAD

**Discussion by meschwartz** [keepassxreboot/keepassxc#10308] [Q&A]
Hi! Back in 2022 at WWDC [Apple announced ScreenCaptureKit](https://developer.apple.com/videos/play/wwdc2022/10156/?time=896). As part of that announcement, they stated:

> With the release of ScreenCaptureKit, older capture frameworks CGDisplayStream and CGWindowList will be deprecated in the future.

KeepassXC currently uses both CGDisplayStream and CGWindowList. Now that it's been almost two years since the announcment, and given how fast Apple moves, it's likely these frameworks will stop working soon, or start behaving differently, as Apple prepares for their deprecation.

Have the developers looked at replacing these frameworks with ScreenCaptureKit?

Thanks!
Mike.

**Comment by droidmonkey** [comment]
We haven't, but deprecation doesn't mean they will remove the capability. A lot of stuff is deprecated in the macOS API for many years. We can look into the other solution, but it may not provide the capabilities we need. We see this very pointedly with Wayland on Linux side.

**Comment by meschwartz** [comment]
I kept my post deliberately vague and coy, to avoid violating any beta NDAs. But I see about an hour after I posted, somebody posted the following on Apple's Developer Forums, and they were considerably less concerned about NDAs :)

[ScreenCaptureKit: new approval dialogs appearing in macOS 14.4 betas, also repeats every time an app launches](https://forums.developer.apple.com/forums/thread/746657)

**Reply by droidmonkey** [reply]
I appreciate that post, thank you.
